cwl/src/test/scala/wdl4s/cwl/ExportCwlSamplesSpec.scala

```diff
@@ -28,27 +39,130 @@ class ExportCwlSamplesSpec extends FlatSpec with Matchers {
           `type` = None
         ))),
         outputs = Coproduct[CommandLineTool.Outputs](Array.empty[CommandOutputParameter]),
-        `class` = "CommandLineTool",
         id = None,
         requirements = None,
         hints = None,
         label = None,
         doc = None,
         cwlVersion = Option(CwlVersion.Version1),
-        baseCommand = None,
+        baseCommand = Option(Coproduct[BaseCommand]("echo")),
         arguments = None,
         stdin = None,
         stderr = None,
         stdout = None,
         successCodes = None,
         temporaryFailCodes = None,
         permanentFailCodes = None)
-    val toolJson = encodeCwlCommandLineTool(tool)
-    val printer = new Printer(preserveOrder = true, dropNullKeys = true, indent = "  ")
-    val toolJsonString = printer.pretty(toolJson)
     val expectedToolJsonString =
-      """{"inputs":{"message":{"inputBinding":{"position":1}}},"outputs":[],"class":"CommandLineTool","cwlVersion":"v1.0"}"""
-    toolJsonString shouldBe expectedToolJsonString
+      """inputs:
+        |  message:
+        |    inputBinding:
+        |      position: 1
+        |outputs: []
+        |class: CommandLineTool
+        |cwlVersion: v1.0
+        |baseCommand: echo
+        |""".stripMargin
+    assertCorrectJson(tool, expectedToolJsonString)
+  }
+
+  it should "encode sample CWL workflow" in {
+    val workflow = Workflow(
+      cwlVersion = Option(CwlVersion.Version1),
+      inputs = Coproduct[WorkflowInput](
+        Map(
+          "inp" -> Coproduct[MyriadInputType](CwlType.File),
+          "ex" -> Coproduct[MyriadInputType](CwlType.String)
+        )
+      ),
+      outputs = Coproduct[WorkflowOutput](
+        Map(
+          "classout" -> WorkflowOutputParameter(
+            id = None,
+            `type` = Option(Coproduct[MyriadOutputType](CwlType.File)),
+            outputSource = Option(Coproduct[WorkflowOutputParameter#OutputSource]("compile/classfile"))
+          )
+        )
+      ),
+      steps = Coproduct[WorkflowSteps](
+        Map(
+          "untar" -> WorkflowStep(
+            id = None,
```

Comment:
For all these places that you're using `None`, could you please make this the default in the class constructor parameter?  Then please omit them in this test.